components/raftstore/src/store/fsm/store.rs

```diff
@@ -221,7 +221,7 @@ pub struct PollContext<T, C: 'static> {
     pub global_stat: GlobalStoreStat,
     pub store_stat: LocalStoreStat,
     pub engines: Engines,
-    pub kv_wb: WriteBatch,
+    pub kv_wb: Option<WriteBatch>,
```

Comment:
Can this be done with a `RefCell` instead? The `Option` leads to a bunch of error-prone unwrap and take patterns here.

Reply:
Actually, this field is accessed from `&mut` methods. It doesn't look to me like it needs to be stored in a wrapper type at all. Why is this an `Option`?

Reply:
Because we can not call two 'mutable reference' function at the same time. The code would not pass compile like this:
```
do_something(ctx.kv_wb_mut(), ctx.raft_wb_mut());
```